poc/flp_generic.py

```diff
@@ -640,6 +641,137 @@ def test_vec_set_type_param(self, test_vec):
         return ['length', 'chunk_length']
 
 
+class MultiHotHistogram(Valid):
+    """
+    A validity circuit that checks each Client's measurement is a bit vector
+    with at most `max_count` number of 1s.
+
+    In order to check the Client measurement `meas` has at most `max_count` 1s,
+    we ask the Client to send an additional `bits_for_count` bits that
+    encode the number of 1s in the measurement, with an offset. Specifically:
+    - Let `bits_for_count = int(log2(max_count)) + 1`.
+    - Let `offset = 2**bits_for_count - 1 - max_count`.
+    - Client will encode `count = offset + \sum_i meas_i` in
+      `bits_for_count` bits.
+    - We can naturally bound `count` as the following:
+      `0 <= count <= 2**bits_for_count - 1`, and therefore:
+      `-offset <= \sum_i meas_i <= max_count`.
+    - Since we also verify each `meas_i` is a bit, we can lower bound the
+      summation by 0. Therefore, we will be able to verify
+      `0 <= \sum_i meas_i <= max_count`.
+    """
+    # Operational parameters
+    length = None  # Set by the constructor
+    max_count = None  # Set by constructor
+    chunk_length = None  # Set by constructor
+
+    # Associated types
+    Measurement = Vec[Unsigned]
```

Comment:
Idea: the measurement is the set of bucket indices.
```suggestion
    Measurement = set[Unsigned]
```

Reply:
I don't have a strong preference.
* Even if you go with `set`, I don't think there's a need to change the API in the DP draft to match: just map one type to the other.
* If you stick with what you have please (1) change `Vec` to `list` and (2) add a comment describing the measurement type.

Reply:
Sounds good. I changed it to `list[Unsigned]` and added comments next to `Measurement` type and `AggResult` type.